Automattic/wp-calypso
# Fix/switch suggested default locale route

#### Context
This PR fixes #61358 and avoids the infinite loading `/log-in/en` discovered only in production after releasing this PR  #63271, which I reverted immediately.

The infinite loading in production was because `page.redirect` doesn't work on the server side.

#### Changes proposed in this Pull Request

- Refactor the login Routes middleware to fix the translation in Login.
- Fix the login redirection when the user clicks on the LocalSuggestion to English. From `/log-in/en` to `/log-in`.
- Check if is server side to avoid using `page.redirect`.

#### Screenshots

**Before**

https://user-images.githubusercontent.com/779993/166646033-b927b1ed-b407-4721-b054-dfb438602e7d.mov


**After**

https://user-images.githubusercontent.com/779993/166646040-dbd2d0be-5ec0-48bb-8d87-729de0cc7ead.mov


#### Testing instructions

1. Your browser should have English as your preferred language.
2. Go to the login screen with any locale different from `en` : [`/log-in/fr`](https://wordpress.com/log-in/fr) 
3. Observe the page is rendered in "french"
4. Click on the Locale Suggestion. It contains this text: `Also available in English`
5. Observe that now the page renders correctly in English, and the URL has removed the `/en` from the URL.
6. The switch to the LocalSuggestion component is in multiple places, but I've found that the issue only happened on the log-in screen.
7. Repeat from step 2 in the jetpack log-in URL : [`/log-in/jetpack/fr`](https://wordpress.com/log-in/jetpack/fr)

- Closes #61358
